src/Data/Singletons/Partition.hs

```diff
@@ -128,3 +142,28 @@ partitionInstanceDec (DLetDec (DFunD name clauses)) =
 partitionInstanceDec (DLetDec (DPragmaD {})) = return Nothing
 partitionInstanceDec _ =
   fail "Only method bodies can be promoted within an instance."
+
+partitionDeriving :: DsMonad m => Maybe DerivStrategy -> DPred -> Maybe DCxt -> DType -> [DCon]
```

Comment:
It seems that this should really return one of *three* possibilities: an instance, a `SDEqDec`, or neither of the above. Perhaps the best way for it to do this is for the function to return a `PartitionedDecs` that can be `<>`ed with other declarations as necessary. That seems to be a cleaner design. (Then, code elsewhere need not detect `Eq` in a `deriving` clause.)